fdbserver/storageserver.actor.cpp

```diff
@@ -2209,10 +2224,17 @@ ACTOR Future<Version> watchWaitForValueChange(StorageServer* data, SpanContext p
 
 	state Version minVersion = data->data().latestVersion;
 	state Future<Void> watchFuture = data->watches.onChange(metadata->key);
+	state Future<Void> tenantWatchFuture;
+	if (tenantId != TenantInfo::INVALID_TENANT) {
+		tenantWatchFuture = data->tenantWatches.onChange(tenantId);
```

Comment:
You could possibly make this more concise by writing `watchFuture = watchFuture || data->tenantWatches.onChange(tenantId)`. Then you might not need some of the changes below.